### DISCUSSION TITLE
`#` in option name?

### DISCUSSION THREAD

**Discussion by alxndr** [nanotee/nvim-lua-guide#66] [Q&A] [answered]
Hi, thanks for documenting so much!!

In [the `vim.fn` section](https://github.com/nanotee/nvim-lua-guide#vimfnfunction) you note that the hash/octothorp character `#` is not valid in Lua identifiers, so vim functions that use it must be called using a string syntax.

Is there something similar available for setting options which use the `#` character?

e.g. [airline](https://github.com/vim-airline/vim-airline) has options such as `let g:airline#extensions#tabline#formatter = 'default'` — how would I do this using `vim.g`?

**Comment by nanotee** [comment, marked as answer]
Hi! I've seen this come up a few times so I've added some clarification in the guide
`vim.g` should work like `vim.fn` here, you can refer to these variables by using table bracket notation: `vim.g['airline#extensions#tabline#formatter'] = 'default'`

**Reply by alxndr** [reply]
Thank you! And thanks again for the guide!
